app/views/dashboard/_activity.html.erb

```diff
@@ -97,16 +84,14 @@
 
   <div class="row">
     <% activity.each_with_index do |node, i| %>
-      <% if node.is_a?(Revision) || node.type == 'page' %>
-        <%= render partial: "dashboard/node_wiki",     locals: { node: node, index: i } %>
-      <% elsif node.is_a?(Comment) %>
+      <% if node.is_a?(Comment) && (params[:sort] == "all" || params[:sort] == "comment") %>
```

Comment:
And then here I think we're having an issue because if the node is a Comment but doesn't meet the sort criteria it is continuing on to checking the next else statement instead of stopping. I suspect we will have to nest the if statements, like this:
```
      <% elsif node.is_a?(Comment) %>
        <% if (params[:sort] == "all" || params[:sort] == "comment") %>
          <%= render partial: "dashboard/node_comment",  locals: { node: node, index: i } %>
        <% end %>
```

Reply:
![FireShot Capture 250 - Action Controller_ Exception caught - localhost](https://user-images.githubusercontent.com/49460529/73458126-c88fef00-4342-11ea-94ad-b1e5739b29bd.png)

You are probably not seeing these errors because you don't have these particular node types in your database. :)

Reply:
@nstjean Thanks for your such a detailed review. I will work upon the issues soon.